plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -648,15 +686,43 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
             // TODO (https://github.com/trinodb/trino/issues/10055) Deny unsupported dates to prevent inserting wrong values. 2106-02-07 is max value in version 20.8
             return WriteMapping.longMapping("Date", dateWriteFunctionUsingLocalDate());
         }
-        if (type == TIMESTAMP_SECONDS) {
-            return WriteMapping.longMapping("DateTime", timestampSecondsWriteFunction());
+        if (type instanceof TimestampType) {
+            TimestampType timestampType = (TimestampType) type;
+            int precision = Math.min(timestampType.getPrecision(), MAX_SUPPORTED_TEMPORAL_PRECISION);
+            String dataType = format("DateTime64(%d)", precision);
+            ClickHouseVersion clickHouseVersion = getClickHouseServerVersion(session);
```

Comment:
Should we cache the server version as a field to avoid remote calls since the server won't likely change once Trino has started?

Reply:
Oh, the method does use a class field, this local var is just for convenience.